src/ImageSharp/Advanced/AotCompiler.cs

```diff
@@ -0,0 +1,103 @@
+    {
+        /// <summary>
+        /// Seeds the compiler using the given pixel format.
+        /// </summary>
+        /// <typeparam name="TPixel">The pixel format.</typeparam>
+        public static void Seed<TPixel>()
+             where TPixel : struct, IPixel<TPixel>
+        {
+            // This is we actually call all the individual methods you need to seed.
+            AotCompileOctreeQuantizer<TPixel>();
+            AotCompileWuQuantizer<TPixel>();
+            AotCompileDithering<TPixel>();
+
+            // TODO: Do the discovery work to figure out what works and what doesn't.
+        }
+
+        /// <summary>
+        /// Seeds the compiler using the given pixel formats.
+        /// </summary>
+        /// <typeparam name="TPixel">The first pixel format.</typeparam>
+        /// <typeparam name="TPixel2">The second pixel format.</typeparam>
+        public static void Seed<TPixel, TPixel2>()
+             where TPixel : struct, IPixel<TPixel>
+             where TPixel2 : struct, IPixel<TPixel2>
+        {
+            Seed<TPixel>();
+            Seed<TPixel2>();
+        }
+
+        /// <summary>
+        /// Seeds the compiler using the given pixel formats.
+        /// </summary>
+        /// <typeparam name="TPixel">The first pixel format.</typeparam>
+        /// <typeparam name="TPixel2">The second pixel format.</typeparam>
+        /// <typeparam name="TPixel3">The third pixel format.</typeparam>
+        public static void Seed<TPixel, TPixel2, TPixel3>()
+             where TPixel : struct, IPixel<TPixel>
+             where TPixel2 : struct, IPixel<TPixel2>
+             where TPixel3 : struct, IPixel<TPixel3>
+        {
+            Seed<TPixel, TPixel2>();
+            Seed<TPixel3>();
+        }
+
+        /// <summary>
+        /// This method doesn't actually do anything but serves an important purpose...
+        /// If you are running ImageSharp on iOS and try to call SaveAsGif, it will throw an excepion:
+        /// "Attempting to JIT compile method... OctreeFrameQuantizer.ConstructPalette... while running in aot-only mode."
+        /// The reason this happens is the SaveAsGif method makes haevy use of generics, which are too confusing for the AoT
+        /// compiler used on Xamarin.iOS. It spins up the JIT compiler to try and figure it out, but that is an illegal op on
+        /// iOS so it bombs out.
+        /// If you are getting the above error, you need to call this method, which will pre-seed the AoT compiler with the
+        /// necessary methods to complete the SaveAsGif call. That's it, otherwise you should NEVER need this method!!!
+        /// </summary>
+        /// <typeparam name="TPixel">The pixel format.</typeparam>
+        private static void AotCompileOctreeQuantizer<TPixel>()
+            where TPixel : struct, IPixel<TPixel>
+        {
+            var test = new OctreeFrameQuantizer<TPixel>(new OctreeQuantizer(false));
+            test.AotGetPalette();
```

Comment:
`GetPalette()` is protected and can't change the access to internal because it is an override. I changed the AotGetPalette method to just call GetPalette instead of duplicating its logic.